src/features/imageEditor/ui/CanvasEditor.tsx

```diff
@@ -0,0 +1,220 @@
+        const { offsetX, offsetY } = event.nativeEvent;
+        const context = contextRef.current;
+        if (context) {
+            context.lineTo(offsetX, offsetY);
+            context.stroke();
+        }
+    };
+
+    const clearCanvas = () => {
+        const canvas = canvasRef.current;
+        const context = contextRef.current;
+        if (canvas && context) {
+            context.clearRect(0, 0, canvas.width, canvas.height);
+        }
+    };
+
+    const drawStar = (cx: number, cy: number, outerX: number, outerY: number) => {
+        const context = contextRef.current;
+        if (!context) return;
+
+        const spikes = 5;
+        const outerRadius = Math.sqrt((outerX - cx) ** 2 + (outerY - cy) ** 2);
+        const innerRadius = outerRadius / 2.5;
+        let rotation = Math.PI / 2 * 3;
+        let x = cx;
+        let y = cy;
+        const step = Math.PI / spikes;
+
+        context.beginPath();
+        for (let i = 0; i < spikes; i++) {
+            x = cx + Math.cos(rotation) * outerRadius;
+            y = cy + Math.sin(rotation) * outerRadius;
+            context.lineTo(x, y);
+            rotation += step;
+
+            x = cx + Math.cos(rotation) * innerRadius;
+            y = cy + Math.sin(rotation) * innerRadius;
+            context.lineTo(x, y);
+            rotation += step;
+        }
+
+        context.lineTo(cx, cy - outerRadius);
+        context.closePath();
+        context.stroke();
+    };
+
+    const drawPolygon = (x1: number, y1: number, x2: number, y2: number, sides: number) => {
+        const context = contextRef.current;
+        if (!context) return;
+
+        const radius = Math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2);
+        const angle = (2 * Math.PI) / sides;
+
+        context.beginPath();
+        for (let i = 0; i < sides; i++) {
+            const x = x1 + radius * Math.cos(angle * i);
+            const y = y1 + radius * Math.sin(angle * i);
+            context.lineTo(x, y);
+        }
+        context.closePath();
+        context.stroke();
+    };
+
+    const drawCircle = (x1: number, y1: number, x2: number, y2: number) => {
+        const context = contextRef.current;
+        if (!context) return;
+
+        const radius = Math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2);
+
+        context.beginPath();
+        context.arc(x1, y1, radius, 0, 2 * Math.PI);
+        context.stroke();
+    };
+
+    const drawRectangle = (x1: number, y1: number, x2: number, y2: number) => {
+        const context = contextRef.current;
+        if (!context) return;
+
+        const width = x2 - x1;
+        const height = y2 - y1;
+
+        context.beginPath();
+        context.rect(x1, y1, width, height);
+        context.stroke();
+    };
+
+    const handleBrushSizeChange = (_: Event, newValue: number | number[]) => {
+        if (typeof newValue === 'number') {
+            setBrushSizeState(newValue);
+            dispatch(setBrushSizeWithState({ brushSize: newValue }));
```

Comment:
Why would you need two states for same thing?

Reply:
You were right! Fix it!